odoo/orm/models.py

```diff
@@ -5835,21 +5843,19 @@ def grouped(self, key):
         :param key: either a callable from a :class:`Model` to a (hashable)
                     value, or a field name. In the latter case, it is equivalent
                     to ``itemgetter(key)`` (aka the named field's value)
-        :type key: callable | str
-        :rtype: dict
         """
         if isinstance(key, str):
             key = itemgetter(key)
 
-        collator = defaultdict(list)
+        collator: defaultdict[typing.Any, list[IdType]] = defaultdict(list)
```

Comment:
Adding a trivial type for a variable used in a simple 4-lines algorithm looks completely ridiculous to me.  Let's stick to method signatures and really non-trivial stuff.  Please don't waste your very valuable brainpower on useless stuff like that :wink: